Sources/SPMBuildCore/PluginInvocation.swift

```diff
@@ -824,6 +824,9 @@ struct PluginScriptRunnerInputSerializer {
         targetFiles.append(contentsOf: try target.underlyingTarget.resources.map {
             .init(basePathId: try serialize(path: $0.path.parentDirectory), name: $0.path.basename, type: .resource)
         })
+        targetFiles.append(contentsOf: try target.underlyingTarget.ignored.map {
```

Comment:
This is the line that actually fixes the bug, which is to include the "ignored" files as source file of unknown role.  Note again that this is different from files marked "excluded" in the manifest — those are never passed to the internal build log and not to plugins either.